### PR TITLE
Fix localized *.intentdefinition have to be added to build source phases

### LINKED ISSUE #614: localised .intentdefinition to sources does not work
While adding localised .intentdefinition file it does not generates source files, because it imports as resource.

directory structure:
```
    |_ App/Sources/Base.lproj/Intents.intentdefinition
    |_ App/Sources/en.lproj/Intents.intentdefinition
    |_ App/Sources/ru.lproj/Intents.intentdefinition
```

project.yml:
```
sources:
   - path: App/Sources
      buildPhase: sources
```

### PR DESCRIPTION
**Pull request by giginet**
[localised .intentdefinition to sources does not work · Issue #614 · yonaskolb/XcodeGen](https://github.com/yonaskolb/XcodeGen/issues/614)

closes #614

# Problem

In the situation,  the project has localized `*.intentdefinition`, they have to be added to `Copy Bundle Resources` phase instead of `Compile Sources` phase.

```
IntentDefinition
├── Base.lproj
│   └── Intents.intentdefinition
├── Info.plist
├── en.lproj
│   └── Intents.strings
└── ja.lproj
    └── Intents.strings
```

Screen Shot 2019-11-19 at 3 17 45

# Description

Localized `*.intentdefinition` should be `Sources`. However, the current implementation treats as `Resources.`

This PR makes the issue to fix.

|Before|After|
|--------|------------|
|||

### CODE REVIEW

File: Sources/XcodeGenKit/SourceGenerator.swift [hunk 1]
```diff
@@ -452,12 +452,10 @@ class SourceGenerator {
                 groupChildren.append(variantGroup)
                 baseLocalisationVariantGroups.append(variantGroup)
 
-                let sourceFile = SourceFile(
-                    path: filePath,
-                    fileReference: variantGroup,
-                    buildFile: PBXBuildFile(file: variantGroup),
-                    buildPhase: .resources
```
Comment (giginet) [inline Sources/XcodeGenKit/SourceGenerator.swift]: Previous implementation, all files under `*.lproj` are treated as `Resources`. However, they should be generated by `generateSourceFile` function.

**Comment by keith** [comment]
Note that depending on what type of intents you're setting up you may prefer this to be in the resources phase and not the compile sources phase because you may not want generated code from this.

**Comment by giginet** [comment]
If you want to treat them as Resources, you can specify `buildPhase`

```yaml
sources:
   - path: Sources/Intents.intentdefinition
     buildPhase: resources
```

File: Sources/XcodeGenKit/SourceGenerator.swift (hunk 1 above)
Comment (brentleyjones) [reply Sources/XcodeGenKit/SourceGenerator.swift]: Should this be it's own PR?

**Comment by giginet** [comment]
Thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@
 - Improved variable expansion runtime [#704](https://github.com/yonaskolb/XcodeGen/pull/704) @rcari
 - Fixed missing headers for static framework targets [#705](https://github.com/yonaskolb/XcodeGen/pull/705) @wag-miles
 - Using more file types from XcodeProj for PBXFileReferences resulting in less project diffs [#715](https://github.com/yonaskolb/XcodeGen/pull/715) @yonaskolb
+- Fixed localized *.intentdefinition have to be added to build source phases [#720](https://github.com/yonaskolb/XcodeGen/pull/720) @giginet
```
Comment (yonaskolb) [inline CHANGELOG.md]: This * can make anything following render incorrectly in Markdown. Please wrap in `

File: Sources/XcodeGenKit/SourceGenerator.swift [hunk 3]
```diff
@@ -81,8 +81,8 @@ class SourceGenerator {
         _ = try getSourceFiles(targetType: .none, targetSource: TargetSource(path: path), path: fullPath)
     }
 
-    func generateSourceFile(targetType: PBXProductType, targetSource: TargetSource, path: Path, buildPhase: TargetSource.BuildPhase? = nil) -> SourceFile {
-        let fileReference = fileReferencesByPath[path.string.lowercased()]!
+    func generateSourceFile(targetType: PBXProductType, targetSource: TargetSource, path: Path, buildPhase: TargetSource.BuildPhase? = nil, fileRefenrece: PBXFileElement? = nil) -> SourceFile {
```
Comment (yonaskolb) [inline Sources/XcodeGenKit/SourceGenerator.swift]: typo `fileRefenrece`

**Comment by kikettas** [comment]
Nicely done @giginet! Looking forward to seeing this in an upcoming release! 🚀

**Comment by yonaskolb** [comment]
The above comments have been resolved on master 👍

**Pull request merged**